symbolic-debuginfo/src/dwarf.rs

```diff
@@ -1248,13 +1265,20 @@ impl<'data> DwarfDebugSession<'data> {
             DwarfInfo::parse(unsafe { &*sections }, symbol_map, address_offset, kind)
         })?;
 
-        Ok(DwarfDebugSession { cell })
+        Ok(DwarfDebugSession {
+            cell,
+            bcsymbolmap: None,
+        })
+    }
+
+    pub(crate) fn set_bcsymbolmap(&mut self, symbolmap: Option<Arc<BcSymbolMap<'data>>>) {
```

Comment:
In the MachO code, this is called `load_symbolmap`. Align the naming and add a doc comment, even though this is not public? 

FWIW, having a dependency on MachO in this implementation is certainly not ideal, but it is the quickest way to go about it. A probably cleaner approach would be to generalize symbol mapping in a trait, but for this one change that may well be an overkill. The last approach dependency-wise would be to wrap the bare DebugSession and transform the `Function` entries we receive back from the session. Since that has it's entirely different problems, though, your solution is about as good as I think it can get now.

Reply:
I tried doing that at first (modifying the `Function` / `FileInfo` structs in the outer iterator), but that would make things a lot more complex actually.